### ISSUE TITLE
Wrong url history for http requests with http proxy in AsyncClient

### ISSUE THREAD

**Issue by oczkers** [encode/httpx#911]
### Checklist
- [X] The bug is reproducible against the latest release and/or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

### Describe the bug
In AsyncClient url in response object is wrong after http request using http proxy - there should be destination url but proxy address is returned instead.
This behavior completely breaks redirects because location header also have proxy url.

EDIT: problem exists only with FORWARD mode

### To reproduce
This is not open proxy, just an example ip so please use some other proxy ip
```python
>>> import httpx
>>> import asyncio
>>> r = httpx.AsyncClient(proxies={'http': 'http://78.47.135.103:8888', 'https': 'http://78.47.135.103:8888'})
>>> asyncio.run(r.get('http://httpbin.org/ip')).url
URL('http://78.47.135.103:8888')
>>> asyncio.run(r.get('https://httpbin.org/ip')).url
URL('https://httpbin.org/ip')
```

### Environment

- OS: Linux
- Python version: 3.7.7
- HTTPX version: 0.12.1 - this is not related to httpcore implementation, exact same behavior was noticed before
 - Async environment: asyncio
- HTTP proxy: yes
- Custom certificates: no

**Comment by mIcHyAmRaNe** [comment]
getting same under httpx: 0.12 with:
```py
import trio
import httpx

proxy = {'http': 'http://181.118.167.104:80', 'https': 'http://181.118.167.104:80'}

async def main():
    async with httpx.AsyncClient(proxies=proxy) as client:
        response = await client.get('https://httpbin.org/ip')
        print(response.url)
        resp = await client.get('http://httpbin.org/ip')
        print(resp.url)

trio.run(main)

## result 
# https://httpbin.org/ip
# http://181.118.167.104:80
```
same code under the pre release httpx: 0.13

<details>
  <summary>Click to expand!</summary>

```sh
Traceback (most recent call last):
  File "test.py", line 23, in <module>
    trio.run(main)
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\trio\_core\_run.py", line 1804, in run
    raise runner.main_task_outcome.error
  File "test.py", line 18, in main
    response = await client.get('https://httpbin.org/ip')
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpx\_client.py", line 1274, in get
    return await self.request(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpx\_client.py", line 1116, in request
    response = await self.send(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpx\_client.py", line 1137, in send
    response = await self.send_handling_redirects(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpx\_client.py", line 1164, in send_handling_redirects
    response = await self.send_handling_auth(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpx\_client.py", line 1201, in send_handling_auth
    response = await self.send_single_request(request, timeout)
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpx\_client.py", line 1233, in send_single_request
    ) = await dispatcher.request(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpcore\_async\http_proxy.py", line 76, in request
    return await self._tunnel_request(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpcore\_async\http_proxy.py", line 147, in _tunnel_request
    proxy_response = await proxy_connection.request(
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpcore\_async\connection.py", line 69, in requ
est
    return await self.connection.request(method, url, headers, stream, timeout)
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpcore\_async\http11.py", line 52, in request
    await self._send_request(method, url, headers, timeout)
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\httpcore\_async\http11.py", line 77, in _send_re
quest
    event = h11.Request(method=method, target=target, headers=headers)
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\h11\_events.py", line 71, in __init__
    self._validate()
  File "C:\Users\Starlab\AppData\Roaming\Python\Python38\site-packages\h11\_events.py", line 143, in _validate
    raise LocalProtocolError("Missing mandatory Host: header")
h11._util.LocalProtocolError: Missing mandatory Host: header
```

</details>

**Comment by sorasful** [comment]
@mIcHyAmRaNe  I tried your snippet with version 0.13.0.dev0 and 0.13.0.dev1  and it worked. And with the good output.

**Comment by mIcHyAmRaNe** [comment]
@sorasful yeah now it's working with `0.13.0.dev1`.
the issue has been resolved with `httpcore>=0.8.3` as tom said here: [#929](https://github.com/encode/httpx/issues/929#issuecomment-624658266)

**Comment by yeraydiazdiaz** [comment]
Thanks @sorasful and @mIcHyAmRaNe, I've also confirmed this issue is resolved as of `0.13.0.dev1`.
